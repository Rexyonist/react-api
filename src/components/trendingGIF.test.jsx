import React from 'react';
import { render, waitFor, screen } from '@testing-library/react';
import MockAdapter from 'axios-mock-adapter';
import TrendGifs from './trendingGIF';

// const axios = require('axios');
// const mock = new MockAdapter(axios);

test('TrendGifs snapshot', () => {
    const {asFragment} = render(<TrendGifs />);
    expect(asFragment()).toMatchSnapshot();
});

// describe('TrendGifs Component', () => {
//     it('renders correctly', async () => {
//         const mockTrendingData = {
//         data: [
//             { id: '1', images: { fixed_height: { url: 'https://example.com/1.gif' } }, title: 'GIF 1' },
//             { id: '2', images: { fixed_height: { url: 'https://example.com/2.gif' } }, title: 'GIF 2' },
//         ],
//         };
//         mock.onGet('https://api.giphy.com/v1/gifs/trending').reply(200, mockTrendingData);

//         render(<TrendGifs />);

//         await waitFor(() => {
//         expect(screen.getByText('Trending GIFs')).toBeInTheDocument();
//         });

//         await waitFor(() => {
//             const gifs = screen.getAllByAltText('GIF');
//             expect(gifs).toHaveLength(2);
//         });
//     });
// });
