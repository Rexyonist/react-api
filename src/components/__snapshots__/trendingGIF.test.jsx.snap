// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrendGifs snapshot 1`] = `
<DocumentFragment>
  <div
    class="trend-gif"
    data-testid="Trend-GIF"
  >
    <h3>
      Trending GIFs
    </h3>
    <div
      class="gif-container"
      data-testid="gif-container"
    >
      <img
        alt="GIF 1"
        src="https://example.com/1.gif"
      />
      <img
        alt="GIF 2"
        src="https://example.com/2.gif"
      />
    </div>
  </div>
</DocumentFragment>
`;
